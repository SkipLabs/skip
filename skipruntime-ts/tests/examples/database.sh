#!/bin/bash
set -e
cd ../../examples

cleanup() {
    jobs -p | xargs kill
}

trap cleanup EXIT

if [ "$#" -lt 2 ]; then
    echo "Usage: $0 <stdout target> <stderr target> <platform>"
fi

if [ "$3" = "native" ]; then
    echo "Running 'database' example on @skipruntime/native"
    LD_LIBRARY_PATH=$(realpath ../../build/skipruntime) SKIP_PLATFORM="native" node dist/database.js >/dev/null &
else
    echo "Running 'database' example on @skipruntime/wasm"
    node dist/database.js >/dev/null &
fi

node dist/database-server.js >/dev/null &
sleep 1 # give a moment for service to spin up
node dist/database-client.js >"$1" 2>"$2"
