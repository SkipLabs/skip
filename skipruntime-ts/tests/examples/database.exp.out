Connect to replication server for resource /users
Init [
  [ '123', [ [Object] ] ],
  [ '124', [ [Object] ] ],
  [ '125', [ [Object] ] ]
]
Set /user/123 to { name: "daniel", country: "UK" }
Update [ [ '123', [ [Object] ] ] ]
Get /user/123 [ { country: 'UK', name: 'daniel' } ]
Delete /user/123
Update [ [ '123', [] ] ]
Get /user/123 []
